App.js

```diff
@@ -0,0 +1,129 @@
+import React, { useState } from "react";
+import {
+  Stack,
+  ThemeProvider,
+  MenuGroup,
+  Flex,
+  Button,
+  Text,
+  IconButton
+} from "@chakra-ui/core";
+import "./App.css";
+import map from "lodash/map";
+import { FiImage } from "react-icons/fi";
+import { TiLocationOutline } from "react-icons/ti";
+import { IoIosPricetag } from "react-icons/io";
+import { GoSettings } from "react-icons/go";
+import { FaBuilding } from "react-icons/fa";
+import Pagination from "./components/pagination.js";
+import DropDown from "./components/header.js";
+function App() {
+  const [offsetValue, setOffsetValue] = useState(0);
+  const [limitValue, setLimitValue] = useState(9);
+  const [pageNumber, setPageNumber] = useState(1);
+  const paginationButtonClick = numberClicked => {
+    setOffsetValue(numberClicked * 9);
+    setLimitValue((numberClicked + 1) * 9);
```

Comment:
This pattern of updating three useState effects for one single Callback will trigger 3 different renders. Use useReducer instead.